pkg/zhttpclient/request.go

```diff
@@ -0,0 +1,84 @@
+package zhttpclient
+
+import (
+	"context"
+	"io"
+	"net/url"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type ZRequest interface {
```

Comment:
Can we add the wrap for SetBody, SetResult and SetError? I think it would be useful. e.g: 
<img width="813" alt="image" src="https://github.com/Zondax/golem/assets/11968535/b2a3a776-4ea7-4b52-a4c8-b594206637bd">